### PR TITLE
feat: create initial `@metamask/multichain` package

### PR DESCRIPTION
**Pull request by jiexi**
## Explanation

Creates the initial (empty) `@metamask/multichain` package to make downstream additions to this package easier to review.

## References

Upstream: https://github.com/MetaMask/core/pull/4811
Downstream: https://github.com/MetaMask/core/pull/4784

### `@metamask/multichain`

- **ADDED**: Initial skeleton package for `@metamask/multichain`

## Checklist

- [X] I've updated the test suite for new or updated code as appropriate
- [X] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [X] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [X] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

**Review by Gudahtt — COMMENTED** [review]
Looks good 👍 Can approve after the base is merged.

**Review by adonesky1 — DISMISSED** [review]
LGTM

File: README.md
Comment (Gudahtt) [inline README.md:150]: Hmm, when I run `yarn update-readme-content`, it restores this.

Maybe the graph changed again since you made this commit

Comment (Gudahtt) [reply README.md:150]: Non-blocking, as I see that this graph is already stale on `main`, but this would be nice to fix

Comment (jiexi) [reply README.md:150]: may have been bad PR splitting on my part

Comment (adonesky1) [reply README.md:150]: Addressed in latest: https://github.com/MetaMask/core/pull/4812/commits/738c29177e159f8465a587bd2de50ba069320c03

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**
